### PR TITLE
loopwm: added updateScript

### PR DESCRIPTION
**Pull request by matteo-pacini**
## Description of changes

- Added `updateScript` for the Loop window manager on Darwin
- Used a plain bash script rather than `nix-update-script` so it works through `r-ryantm` bot (similar issue to this #329752 )

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by s0me1newithhand7s — APPROVED** [review]

**Comment by matteo-pacini** [comment]
Friendly ping for a merge 🙏🏻  @toonn @K900 @SuperSandro2000 @NickCao

File: pkgs/by-name/lo/loopwm/update.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p curl jq gnused
+
+set -euo pipefail
+
+ROOT="$(dirname "$(readlink -f "$0")")"
+if [[ ! "$(basename "$ROOT")" == "loopwm" || ! -f "$ROOT/package.nix" ]]; then
```
Comment (SuperSandro2000) [inline pkgs/by-name/lo/loopwm/update.sh:7]: Why not replace this with cd $(basename $0) ?

File: pkgs/by-name/lo/loopwm/update.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+replace_version_in_file() {
+    # $1: file
+    # $2: new version
+    sed -i "s#version = \".*\";#version = \"$2\";#g" "$1"
+}
+
+LOOPWM_DARWIN_HASH="$(get_hash "https://github.com/MrKai77/Loop/releases/download/${LOOPWM_LATEST_VER}/Loop.zip")"
+
+replace_hash_in_file "$PACKAGE_NIX" "$LOOPWM_DARWIN_HASH"
```
Comment (SuperSandro2000) [inline pkgs/by-name/lo/loopwm/update.sh:42]: We don't need to complicate things, we can just inline this and short the script with it

**Comment by matteo-pacini** [comment]
@SuperSandro2000 thanks for the feedback, simplified the script as per your suggestions - should be clearer now!

**Review by toonn — APPROVED** [review]
LGTM

**Comment by matteo-pacini** [comment]
@SuperSandro2000 @toonn friendly ping for a merge 🙏🏻

**Pull request merged**